samples/nrf9160/modem_shell/src/ltelc/ltelc_api.c

```diff
@@ -0,0 +1,944 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <assert.h>
+
+#include <sys/types.h>
+#include <shell/shell.h>
+
+#include <modem/modem_info.h>
+
+#include <posix/arpa/inet.h>
+#include <net/net_ip.h>
+
+#include "utils/net_utils.h"
+
+#include "ltelc_shell.h"
+#include "ltelc_shell_print.h"
+#include "ltelc_api.h"
+
+#if defined(CONFIG_AT_CMD)
+
+#include <modem/at_cmd.h>
+#include <modem/at_cmd_parser.h>
+#include <modem/at_params.h>
+
+#define AT_CMD_BUFFER_LEN (CONFIG_AT_CMD_RESPONSE_MAX_LEN + 1)
+#define AT_CMD_PDP_CONTEXTS_READ "AT+CGDCONT?"
+#define AT_CMD_PDP_CONTEXTS_READ_PARAM_COUNT 12
+#define AT_CMD_PDP_CONTEXTS_READ_CID_INDEX 1
+#define AT_CMD_PDP_CONTEXTS_READ_PDP_TYPE_INDEX 2
+#define AT_CMD_PDP_CONTEXTS_READ_APN_INDEX 3
+#define AT_CMD_PDP_CONTEXTS_READ_PDP_ADDR_INDEX 4
+
+#define AT_CMD_PDP_CONTEXT_READ_INFO \
+	"AT+CGCONTRDP=%d" /* Use sprintf to add CID into command */
+#define AT_CMD_PDP_CONTEXT_READ_INFO_PARAM_COUNT 20
+#define AT_CMD_PDP_CONTEXT_READ_INFO_CID_INDEX 1
+#define AT_CMD_PDP_CONTEXT_READ_INFO_DNS_ADDR_PRIMARY_INDEX 6
+#define AT_CMD_PDP_CONTEXT_READ_INFO_DNS_ADDR_SECONDARY_INDEX 7
+#define AT_CMD_PDP_CONTEXT_READ_INFO_MTU_INDEX 12
+
+#define AT_CMD_PDP_CONTEXT_READ_RSP_DELIM "\r\n"
+
+static int ltelc_api_pdn_id_get(uint8_t cid)
+{
+	int ret;
+	char *p;
+	char resp[32];
+	static char at_buf[32];
+
+	errno = 0;
+
+	ret = snprintf(at_buf, sizeof(at_buf), "AT%%XGETPDNID=%u", cid);
+	if (ret < 0 || ret >= sizeof(at_buf)) {
+		return -ENOBUFS;
+	}
+
+	ret = at_cmd_write(at_buf, resp, sizeof(resp), NULL);
+	if (ret) {
+		return ret;
+	}
+
+	p = strchr(resp, ':');
+	if (!p) {
+		return -EBADMSG;
+	}
+	ret = strtoul(p + 1, NULL, 10);
+	if (errno) {
+		ret = -errno;
+	}
+
+	return ret;
+}
+
+#define AT_CMD_PDP_CONTEXTS_ACT_READ "AT+CGACT?"
+
+static void ltelc_api_get_pdn_states(struct pdp_context_info_array_t *pdp_info)
+{
+	char buf[16] = { 0 };
+	const char *p;
+	char at_response_str[256];
+	int ret;
+	int ctx_cnt = pdp_info->size;
+	struct pdp_context_info_t *ctx_tbl = pdp_info->array;
+
+	ret = at_cmd_write(AT_CMD_PDP_CONTEXTS_ACT_READ, at_response_str,
+			   sizeof(at_response_str), NULL);
+	if (ret) {
+		printk("Cannot get PDP contexts activation states, err: %d",
```

Comment:
You should rather use logger instead - it integrates with shell nicely (there's a shell backed for the logger).

Reply:
I've changed all these printf's to shell_print/shell_error throughout MOSH. However, there are some that are call from main() which are left as printk as shell is not ready to print necessarily.